raft/raft.go

```diff
@@ -911,11 +911,31 @@ func (r *raft) Step(m pb.Message) error {
 				r.logger.Warningf("%x is unpromotable and can not campaign; ignoring MsgHup", r.id)
 				return nil
 			}
-			ents, err := r.raftLog.slice(r.raftLog.applied+1, r.raftLog.committed+1, noLimit)
-			if err != nil {
-				r.logger.Panicf("unexpected error getting unapplied entries (%v)", err)
+
+			n := 0
+
+			// Trying to retrieve all entries at once might cause out-of-memory issues in
+			// cases where the raft log is too big to fit into memory. Instead of retrieving
+			// all entries at once, retrieve it in batches of 64MB.
+			for batchFirst := r.raftLog.applied + 1; batchFirst <= r.raftLog.committed; {
+				ents, err := r.raftLog.slice(batchFirst, r.raftLog.committed+1, 64<<20)
```

Comment:
Can we define `64<<20` as a constant with comment?

Reply:
Maybe this should use Config.MaxCommittedSizePerReady instead.

Reply:
> Maybe this should use Config.MaxCommittedSizePerReady instead.

Config.MaxCommittedSizePerReady is set to 1 * 1024 * 1024(1m) default, that may be a little small batch.